base_layer/core/src/blocks/blockheader.rs

```diff
@@ -150,7 +150,15 @@ impl BlockHeader {
 
     /// Calculates and returns the achieved difficulty for this header and associated proof of work.
     pub fn achieved_difficulty(&self) -> Difficulty {
-        self.pow.achieved_difficulty(self)
+        ProofOfWork::achieved_difficulty(self)
+    }
+
+    /// Calculates the total accumulated difficulty for the blockchain from the genesis block up until (and including)
+    /// this block.
+    pub fn total_accumulated_with_achieved_difficulty(&self) -> Difficulty {
```

Comment:
```suggestion
    pub fn total_accumulated_difficulty_inclusive(&self) -> Difficulty {
```

Your choice. See which sounds better

Reply:
Changed to `total_accumulated_difficulty_inclusive `.